File: app/src/main/java/com/lucas/omnia/models/Comment.java
 package com.lucas.omnia.models;

/**
 * Created by Lucas on 21/11/2017.
 */

 import com.google.firebase.database.Exclude;
 import com.google.firebase.database.IgnoreExtraProperties;

 import java.util.HashMap;
 import java.util.Map;

 @IgnoreExtraProperties
 public class Comment {

     public String uid;
     public String author;
     public long timestamp;
     public String body;
     public boolean edited;
     public int upVoteCount = 0;
     public int downVoteCount = 0;
     public int replyCount = 0;
     public Map<String, Boolean> upVotes = new HashMap<>();
     public Map<String, Boolean> downVotes = new HashMap<>();
     public Map<String, Boolean> replies = new HashMap<>();

     public Comment() {
         // Default constructor required for calls to DataSnapshot.getValue(Comment.class)
     }

     public Comment(String uid, String author, String body) {
         this.uid = uid;
         this.author = author;
         this.body = body;
     }

     @Exclude
     public Map<String, Object> toMap() {
         HashMap<String, Object> result = new HashMap<>();
         result.put("uid", uid);
         result.put("author", author);
         result.put("timestamp", timestamp);
         result.put("body", body);
         result.put("edited", edited);
         result.put("upVoteCount", upVoteCount);
         result.put("downVoteCount", downVoteCount);
         result.put("replyCount", replyCount);
         result.put("upVotes", upVotes);
         result.put("downVotes", downVotes);
         result.put("replies", replies);

         return result;
     }
 }

File: app/src/main/java/com/lucas/omnia/models/Comment.kt
package com.lucas.omnia.models

import com.google.firebase.database.Exclude
import com.google.firebase.database.IgnoreExtraProperties
import java.util.*

@IgnoreExtraProperties
class Comment {
    var uid: String? = null
    var author: String? = null
    var timestamp: Long = 0
    var body: String? = null
    var edited = false
    var upVoteCount = 0
    var downVoteCount = 0
    var votesBalance = 0
    var replyCount = 0
    var upVotes: Map<String, Boolean> = HashMap()
    var downVotes: Map<String, Boolean> = HashMap()

    constructor() {
        // Default constructor required for calls to DataSnapshot.getValue(Comment.class)
    }

    constructor(uid: String?, author: String?, body: String?) {
        this.uid = uid
        this.author = author
        this.body = body
    }

    @Exclude
    fun toMap(): Map<String, Any?> {
        val result = HashMap<String, Any?>()
        result["uid"] = uid
        result["author"] = author
        result["timestamp"] = timestamp
        result["body"] = body
        result["upVoteCount"] = upVoteCount
        result["downVoteCount"] = downVoteCount
        result["votesBalance"] = votesBalance
        result["replyCount"] = replyCount
        return result
    }
}